### PR TITLE
Improved 68340, 68681 and dbox support

### PR DESCRIPTION
**Pull request by JoakimLarsson**
A cleaner PR than [my previous](https://github.com/mamedev/mame/pull/2596). I reached my mile stone as I am able to hook up a terminal to the serial port. In theory it will now be possible to flash the dbox with a shareware called DVB2000 by using a PC utility that connects over serial, it is not tested yet as I need to figure out how to do that, either from a DOSbox or a second MAME instance. Please review in particular the way I created the serial module by inherit the duart driver. I think it works well but there could impurities in the way I did it.

### CODE REVIEW

**Comment by DavidHaywood** [comment]
this should come in handy with the MPU5 driver, various things connect to the internal serial on that.

**Comment by JoakimLarsson** [comment]
I probably need some help with this, It is not clear to me what makes -validate sour, can someone please explain the problem in layman terms or I need to read up on what -validate is trying to do which can take some time.

**Comment by Tafoid** [comment]
All I know is the it appears two devices are being detected as having shared setname/descriptions.  They also seem to be in separate drivers or possibly being miss-reported.  The -listxml output only shows mc68681.cpp as a driver source for both devices, so there may be an issue with detection.

src/devices/machine/mc68681.cpp
src/devices/machine/68340.cpp

**Comment by cuavas** [comment]
OK, there are a whole lot of problems here:
* src/devices/machine/68340ser.cpp:171 is clearly not going to work.  It has spaces in the short name "m68340 serial module".  Remember device short names have similar rules to driver names (they have to be used for file names, on command line and as database keys).
* The full names "M68340 SERIAL MODULE" and "Motorola 68340 Serial Module" are confusingly similar.  What's different about the derived device?  Give it a name that makes what it is apparent.
* Having classes `m68340_serial_device` and `m68340_serial` is going to confuse people, particularly when they're both `device_t` implementations.  We conventionally put the `_device` suffix on names of `device_t` implementations, so whatever you call the derived one should still have it.  But the name needs to clearly differentiate it from the base one in a meaningful way (i.e. so someone looking at the names has a decent chance of knowing what the differences are).
* The constructor for the derived `m68340_serial` calls the public constructor of `m68340_serial_device` so it doesn't supply its `device_type` resulting in instances reporting the wrong type.  Look at any other `device_` implementation with derived variants to see the way to fix this.

So to fix this, you need to:
* Make the names adhere to rules/conventions
* Make the differences between the base/derived devices apparent from the names
* Make the derived device supply its own type in its constructor

**Comment by JoakimLarsson** [comment]
Thanks a lot, I will read this at least twice slowly and apply the changes needed, I was mainly looking for a "duplicate" problem, as indicated by -validate, which I could not find, your comments are quite a bit more detailed. I was not happy with the naming either but as they compiled and ran as expected I thought they were fine.
